fix compilation with cmake 2.8.4 by backporting some changes from 3.8.9 and applying
	https://awesome.naquadah.org/bugs/index.php?do=details&task_id=869

https://bugs.gentoo.org/show_bug.cgi?id=356009
--- awesome-3.4.4-orig/CMakeLists.txt	2011-02-22 15:26:17.380000002 +0200
+++ CMakeLists.txt	2011-02-22 16:03:40.219000002 +0200
@@ -249,22 +249,24 @@
 # }}}
 
 # {{{ Theme icons
-file(GLOB icon_sources ${SOURCE_DIR}/themes/*/titlebar/*.png)
-set(ALL_ICONS ${icon_sources})
+file(GLOB icon_sources RELATIVE ${SOURCE_DIR} ${SOURCE_DIR}/themes/*/titlebar/*.png)
+
+foreach(icon ${icon_sources})
+    # Copy all icons to the build dir to simplify the following code.
+    # Source paths are interpreted relative to ${SOURCE_DIR}, target paths
+    # relative to ${BUILD_DIR}.
+    get_filename_component(icon_path ${icon} PATH)
+    get_filename_component(icon_name ${icon} NAME)
+    file(COPY ${icon} DESTINATION ${icon_path})
+    set(ALL_ICONS ${ALL_ICONS} "${icon_path}/${icon_name}")
+endforeach()
 
 macro(a_icon_convert match replacement input)
     string(REPLACE ${match} ${replacement} output ${input})
 
     if(NOT ${input} STREQUAL ${output})
-        if(NOT ${output} MATCHES ${BUILD_DIR})
-            string(REPLACE ${SOURCE_DIR} ${BUILD_DIR} output ${output})
-        endif()
-
         set(ALL_ICONS ${ALL_ICONS} ${output})
 
-        get_filename_component(output_path ${output} PATH)
-        file(MAKE_DIRECTORY ${output_path})
-
         add_custom_command(
             COMMAND ${CONVERT_EXECUTABLE} ${input} ${ARGN} ${output}
             OUTPUT  ${output}
